subprojects/docs/src/samples/userguide/artifacts/defineRepository/build.gradle

```diff
@@ -140,6 +140,20 @@ repositories {
 }
 //END SNIPPET authenticated-maven-repo
 
+//START SNIPPET header-authenticated-maven-repo
+repositories {
+    maven {
+        url "http://repo.mycompany.com/maven2"
+        credentials(HttpHeaderCredentials) {
+            header "Private-Token: TOKEN"
```

Comment:
I'm still wondering whether `header("Private-Token", "TOKEN")` would be the better API. Thoughts?

Reply:
Yes, I agree. Didn't get it to work though. I'll try again.

Reply:
How about `void header(String name, String value) { ... }`?

Reply:
Or, probably even better, what about the following?

```gradle
credentials(HttpHeaderCredentials) {
	name = "Private-Token"
	value = "TOKEN"
}
```

Reply:
@paulwellnerbou Let's go with the last one, shall we?

Reply:
@marcphilipp, already implementing it right now :)